providers/hubspot/metadata.go

```diff
@@ -498,3 +506,59 @@ type stage struct {
 	Value       string `json:"id"`
 	DisplayName string `json:"label"`
 }
+
+// fetchRequiredFieldsBestEffort fetches the object's schema and marks required fields.
+// If the schema cannot be fetched due to missing scopes, it returns the original
+// fields unchanged without an error. Other errors are returned normally.
+func (c *Connector) fetchRequiredFieldsBestEffort(
+	ctx context.Context, objectName string, fields map[string]common.FieldMetadata,
+) (map[string]common.FieldMetadata, error) {
+	url, err := c.getObjectSchemaURL(objectName)
+	if err != nil {
+		return nil, err
+	}
+
+	rsp, err := c.Client.Get(ctx, url.String())
+	if err != nil {
+		if isMissingSchemasScope(err) {
+			// User does not have permission to access the schema endpoint.
+			// Return the original fields without enrichment.
+			return fields, nil
```

Comment:
can you add a log line, something like

```
logging.VerboseLogger(ctx).Debug(fmt.Sprintf("Not populating isRequired for fields of %s because scopes are missing", objectName))
```